pkg/controller/manager/manager_controller.go

```diff
@@ -285,24 +285,28 @@ func (r *ReconcileManager) Reconcile(ctx context.Context, request reconcile.Requ
 		certOperatorManaged = utils.IsOperatorIssued(issuer)
 	}
 
-	// If the secret does not exist, then create one.
-	// If the secret exists but is operator managed, then check that it has the
-	// right DNS names and update it if necessary.
-	if tlsSecret == nil || certOperatorManaged {
-		// Create the cert if doesn't exist. If the cert exists, check that the cert
-		// has the expected DNS names. If the cert doesn't exist, the cert is recreated and returned.
-		// Note that validation of DNS names is not required for a user-provided manager TLS secret.
-		svcDNSNames := dns.GetServiceDNSNames(render.ManagerServiceName, render.ManagerNamespace, r.clusterDomain)
-		svcDNSNames = append(svcDNSNames, "localhost")
-		certDur := 825 * 24 * time.Hour // 825days*24hours: Create cert with a max expiration that macOS 10.15 will accept
-		tlsSecret, err = utils.EnsureCertificateSecret(
-			render.ManagerTLSSecretName, tlsSecret, render.ManagerSecretKeyName, render.ManagerSecretCertName, certDur, svcDNSNames...,
-		)
-
-		if err != nil {
-			r.status.SetDegraded(fmt.Sprintf("Error ensuring manager TLS certificate %q exists and has valid DNS names", render.ManagerTLSSecretName), err.Error())
-			return reconcile.Result{}, err
+	if installation.CertificateManagement == nil {
+		// If the secret does not exist, then create one.
+		// If the secret exists but is operator managed, then check that it has the
+		// right DNS names and update it if necessary.
+		if tlsSecret == nil || certOperatorManaged {
+			// Create the cert if doesn't exist. If the cert exists, check that the cert
+			// has the expected DNS names. If the cert doesn't exist, the cert is recreated and returned.
+			// Note that validation of DNS names is not required for a user-provided manager TLS secret.
+			svcDNSNames := dns.GetServiceDNSNames(render.ManagerServiceName, render.ManagerNamespace, r.clusterDomain)
+			svcDNSNames = append(svcDNSNames, "localhost")
+			certDur := 825 * 24 * time.Hour // 825days*24hours: Create cert with a max expiration that macOS 10.15 will accept
+			tlsSecret, err = utils.EnsureCertificateSecret(
+				render.ManagerTLSSecretName, tlsSecret, render.ManagerSecretKeyName, render.ManagerSecretCertName, certDur, svcDNSNames...,
+			)
+
+			if err != nil {
+				r.status.SetDegraded(fmt.Sprintf("Error ensuring manager TLS certificate %q exists and has valid DNS names", render.ManagerTLSSecretName), err.Error())
+				return reconcile.Result{}, err
+			}
 		}
+	} else if certOperatorManaged {
+		tlsSecret = nil
 	}
```

Comment:
I think there should be an else case here for if there is a user provided cert and CertificateManagement defined, it should result in a degraded state.

Reply:
For the time being the behaviour is that if the user brings his own TLS secret, we default to the old behaviour and mount that BYO secret. This is also part of the test cycles of the feature.

Currently we have not budgetted the time to improve this, but ideally we allow for the CRD to let a user define what SANs we should add to the TLS secret of the manager. This should be picked up in a next iteration on cert management.

Reply:
Was this covered in a design doc that I'm forgetting about? What I'm talking about is the user facing API behavior.
>Currently we have not budgetted the time to improve this, but ideally we allow for the CRD to let a user define what SANs we should add to the TLS secret of the manager. This should be picked up in a next iteration on cert management.

I didn't understand the relevance at first, but I think it is because the user may want to use CertificateManagement but they need to be able to provide SANs for the manager but that isn't possible so we're allowing specifying a TLS secret and CertificateManagement?
It seems like a partially implemented feature then.

I don't like this because that means we'll need to change behavior when we have this fully implemented where SANs can be specified for CertificateManagement. My assumption is that when we have the feature implemented fully, we will want to degrade if both the manager TLS secret and CertificateManagement are specified. That is not how this is currently implemented though.
I think my preference would be to go ahead and degrade in this case and say if you need to specify SANs for the manager then you cannot use CertificateManagement yet.